modules/attachment/variables.tf

```diff
@@ -1,6 +1,9 @@
-variable "profile_id" {
-  type        = string
-  description = "ID of the profile you wish to use for the attachment."
+variable "profile" {
+  type = object({
+    profile_name    = string
+    profile_version = string
```

Comment:
perhaps shorten to name and version attributes since the variable itself is called profile